File: Solutions/349. Intersection of Two Arrays/349. Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> s(nums1.begin(),nums1.end());
        vector<int> res;
        for(auto& x:nums2)
            if(s.erase(x))
                res.emplace_back(x);
        return res;
    }
};

File: Solutions/349. Intersection of Two Arrays/349. Intersection of Two Arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        var s=new HashSet<Integer>();
        for(int i=0;i<nums1.length;++i)
            s.add(nums1[i]);
        List<Integer> l=new ArrayList<>();
        for(var i:nums2)
            if(s.remove(i)==true)
                l.add(i);
        int[] res=new int[l.size()];
        for(int i=0;i<l.size();++i)
            res[i]=l.get(i);
        return res;
    }
}